src/libraries/System.Private.CoreLib/src/System/Number.BigInteger.cs

```diff
@@ -602,6 +602,11 @@ public static void DivRem(ref BigInteger lhs, ref BigInteger rhs, out BigInteger
                         {
                             remLength--;
                         }
+                        else
+                        {
+                            // As soon as we find a non-zero block, the rest of remainder is significant
+                            break;
+                        }
```

Comment:
This change has its own separate PR here: https://github.com/dotnet/runtime/pull/72115